lib/galaxy/webapps/galaxy/api/histories.py

```diff
@@ -28,16 +63,338 @@
+        return self.service.published(trans, serialization_params, filter_params)
+
+    @router.get(
+        '/api/histories/shared_with_me',
+        summary='Return all histories that are shared with the current user.',
+    )
+    def shared_with_me(
+        self,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+        serialization_params: SerializationParams = Depends(query_serialization_params),
+        filter_params: HistoryFilterQueryParams = Depends(HistoryFilterQueryParams),
+    ) -> List[AnyHistoryView]:
+        return self.service.shared_with_me(trans, serialization_params, filter_params)
+
+    @router.get(
+        '/api/histories/most_recently_used',
+        summary='Returns the most recently used history of the user.',
+    )
+    def show_recent(
+        self,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+        serialization_params: SerializationParams = Depends(query_serialization_params),
+    ) -> AnyHistoryView:
+        return self.service.show(trans, serialization_params)
+
+    @router.get(
+        '/api/histories/{id}',
+        summary='Returns the history with the given ID.',
+    )
+    def show(
+        self,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+        id: EncodedDatabaseIdField = HistoryIDPathParam,
+        serialization_params: SerializationParams = Depends(query_serialization_params),
+    ) -> AnyHistoryView:
+        return self.service.show(trans, serialization_params, id)
+
+    @router.get(
+        '/api/histories/{id}/citations',
+        summary='Return all the citations for the tools used to produce the datasets in the history.',
+    )
+    def citations(
+        self,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+        id: EncodedDatabaseIdField = HistoryIDPathParam,
+    ) -> List[Any]:
+        return self.service.citations(trans, id)
+
+    @router.post(
+        '/api/histories',
+        summary='Creates a new history.',
+    )
+    async def create(
+        self,
+        request: Request,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+        payload: CreateHistoryPayload = Depends(CreateHistoryFormData.as_form),
+        serialization_params: SerializationParams = Depends(query_serialization_params),
+    ) -> Union[JobImportHistoryResponse, AnyHistoryView]:
+        """The new history can also be copied form a existing history or imported from an archive or URL."""
+        # This action needs to work both with json and x-www-form-urlencoded payloads.
+        # The way to support different content types on the same path operation is reading
+        # the request directly and parse it depending on the content type.
+        # We will assume x-www-form-urlencoded by default to deal with possible file uploads.
+        # See https://github.com/tiangolo/fastapi/issues/990#issuecomment-639615888
+        if "application/json" in request.headers.get("content-type", ""):
+            body = await request.json()
```

Comment:
Is it mandatory to use await here ? According to the docs we shouldn't create async routes as long as we do IO (which we do by creating the history in the database) ... not quite sure what the implications are since the sync code runs in a threadpool, but if it is possible to make this a sync method I think we should do that.

Reply:
And if we need to use `await`, could we make this an async path dependency ?

Reply:
Good point! It is mandatory to use await as the `request.json()` function returns a coroutine, but I didn't think about using an async path dependency, so thank you very much for the hint!